src/lightning/pytorch/trainer/connectors/data_connector.py

```diff
@@ -450,6 +450,13 @@ def _worker_check(trainer: "pl.Trainer", using_spawn: bool, dataloader: object,
             category=PossibleUserWarning,
         )
 
+    if dataloader.persistent_workers and dataloader.pin_memory and trainer.reload_dataloaders_every_n_epochs > 0:
+        rank_zero_warn(
+            "The combination of `DataLoader(`pin_memory=True`, `persistent_workers=True`) and `Trainer("
+            "reload_dataloaders_every_n_epochs > 0)` can lead to instability due to limitations in PyTorch"
+            "(https://github.com/pytorch/pytorch/issues/91252). We recommend setting `pin_memory=False` in this case."
+        )
```

Comment:
I propose to add a warning to resolve #18487 for the following reasons:
- The limitation on PyTorch seems difficult to solve and if it is ever solved, it would be a while until it becomes our minimum supported PyTorch version.
- The crash (too many files open) could happen far into a long training run, and users typically point fingers towards Lightning first. We should mitigate the impact this annoying limitation has on users. 
 
The concern here is that users generally ignore warnings. We could also raise an error instead, and I would be ok with it but some might say this is too radical. Please share your opinion.